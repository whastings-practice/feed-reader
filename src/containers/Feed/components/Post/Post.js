import React from 'react';

export default function Post({ post }) {
  return (
    <article>
      <h3>
        <a href={post.link} target="_blank" rel="noopener">{post.title}</a>
      </h3>
      {post.publishedAt && <time dateTime={post.publishedAt}>{post.publishedAtFormatted}</time>}
      <div dangerouslySetInnerHTML={{ __html: post.description }} />
    </article>
  );
}
