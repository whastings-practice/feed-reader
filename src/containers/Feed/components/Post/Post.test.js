import React from 'react';
import { shallow } from 'enzyme';

import Post from './Post';

describe('Post', () => {
  let props;
  let wrapper;

  function render() {
    wrapper = shallow(<Post {...props} />);
  }

  beforeEach(() => {
    props = {
      post: {
        title: 'Foo',
        link: '/foo',
        description: '<p>Foo!</p>',
      },
    };
    render();
  });

  test('it renders a post title and description', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it links title to the post link', () => {
    const link = wrapper.find('a');
    expect(link).toHaveProp('href', props.post.link);
    expect(link).toHaveProp('target', '_blank');
    expect(link).toHaveProp('rel', 'noopener');
    expect(link).toHaveText(props.post.title);
  });

  describe('when post has published at info', () => {
    beforeEach(() => {
      Object.assign(props.post, {
        publishedAt: '2018-01-01',
        publishedAtFormatted: 'January 1, 2018',
      });
      render();
    });

    test('it renders the info in a <time> element', () => {
      const timeEl = wrapper.find('time');
      expect(timeEl).toHaveProp('dateTime', props.post.publishedAt);
      expect(timeEl).toHaveText(props.post.publishedAtFormatted);
    });
  });
});
