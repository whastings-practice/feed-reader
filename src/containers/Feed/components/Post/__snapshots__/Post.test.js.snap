// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post it renders a post title and description 1`] = `
<article>
  <h3>
    <a
      href="/foo"
      rel="noopener"
      target="_blank"
    >
      Foo
    </a>
  </h3>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Foo!</p>",
      }
    }
  />
</article>
`;
